x-pack/packages/ml/aiops_log_rate_analysis/queries/fetch_mini_histograms_for_significant_groups.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+import type { Logger } from '@kbn/logging';
+import type {
+  NumericChartData,
+  SignificantItemGroup,
+  SignificantItemGroupHistogram,
+} from '@kbn/ml-agg-utils';
+import { createRandomSamplerWrapper } from '@kbn/ml-random-sampler-utils';
+import { isRequestAbortedError } from '@kbn/aiops-common/is_request_aborted_error';
+
+import { RANDOM_SAMPLER_SEED } from '../constants';
+import type { AiopsLogRateAnalysisSchema } from '../api/schema';
+
+import { getGroupFilter } from './get_group_filter';
+import { getHistogramQuery } from './get_histogram_query';
+import {
+  getMiniHistogramDataFromAggResponse,
+  getMiniHistogramAgg,
+  MINI_HISTOGRAM_AGG_PREFIX,
+  type MiniHistogramAgg,
+} from './mini_histogram_utils';
+
+export const fetchMiniHistogramsForSignificantGroups = async (
+  esClient: ElasticsearchClient,
+  params: AiopsLogRateAnalysisSchema,
+  significantGroups: SignificantItemGroup[],
+  overallTimeSeries: NumericChartData['data'],
+  logger: Logger,
+  // The default value of 1 means no sampling will be used
```

Comment:
Just a question - should we default to no sampling? Is there a chance this could set up users to inadvertently have performance issues?